### PR TITLE
Better stopgaps for log1p and expm1

### PR DESCRIPTION
**Pull request by lrnv**
Hey, 

Thanks a lot for what you did a while back about these stopgaps: this improved a lot my workflow and my runtime ! However i saw that the only two that i am calling these days are log1p and exp1m, and those might not need BigFloat at all.

This two functions are easier to remove from the BigFloat stopgaps than others, even if the algorithms are not the expected ones the result should still be OK, and way faster.

This should NOT be the final implementation: these functions exists because of catastrophic cancellation, and they should be re-implmeented properly. But in the mean time, these are way better than the BigFloat stopgaps.

### CODE REVIEW

File: src/MultiFloats.jl [hunk 1]
```diff
@@ -495,10 +495,19 @@ function Base.log(x::MultiFloat{T,N}) where {T,N}
     return y
 end
 
+function Base.log1p(x::MultiFloat{T,N}) where {T,N}
+    y = x+1
+    return Base.log(y)
+end
+
+function Base.expm1(x::MultiFloat{T,N}) where {T,N}
+    return Base.exp(x)-1
```
Comment (oscardssmith) [inline src/MultiFloats.jl:504]: This could be written as
```
t = tanh(x/2)
return 2*t/(1-t)
```

Comment (oscardssmith) [reply src/MultiFloats.jl:504]: This identity will give a relatively efficient implementation that should be accurate to 3 ULP or so.

File: src/MultiFloats.jl [hunk 2]
```diff
@@ -495,10 +495,19 @@ function Base.log(x::MultiFloat{T,N}) where {T,N}
     return y
 end
 
+function Base.log1p(x::MultiFloat{T,N}) where {T,N}
+    y = x+1
+    return Base.log(y)
```
Comment (oscardssmith) [inline src/MultiFloats.jl:500]: This should be `2*atanh(x/(2+x))` for better accuracy and similar speed.

**Review by oscardssmith — APPROVED** [review]

File: src/MultiFloats.jl (hunk 1 above)
Comment (lrnv) [reply src/MultiFloats.jl]: Done

File: src/MultiFloats.jl (hunk 2 above)
Comment (lrnv) [reply src/MultiFloats.jl]: Done

**Comment by lrnv** [comment]
Thanks @oscardssmith :)

**Pull request merged**